StratisMasternodeDashboard-master/Entities/PendingPoll.cs

```diff
@@ -45,6 +45,49 @@ public string Hash
                 return hashTokens.Length < 2 ? string.Empty : hashTokens[1].Replace("'", string.Empty);
             }
         }
+        [JsonIgnore]
+        public string PubKey
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(this.VotingDataString)) return string.Empty;
+                string[] tokens = this.VotingDataString.Split(',');
+                if (tokens.Length < 1) return string.Empty;
+                string hashToken = tokens.FirstOrDefault(t => t.Contains("pubkey", StringComparison.OrdinalIgnoreCase));
+                if (hashToken == null) return string.Empty;
+                string[] hashTokens = hashToken.Split(':');
+                return hashTokens.Length < 2 ? string.Empty : hashTokens[2].Replace("'", string.Empty);
+            }
+        }
+        [JsonIgnore]
+        public string CollateralAmount
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(this.VotingDataString)) return string.Empty;
+                string[] tokens = this.VotingDataString.Split(',');
+                if (tokens.Length < 1) return string.Empty;
+                string hashToken = tokens.FirstOrDefault(t => t.Contains("collateralamount", StringComparison.OrdinalIgnoreCase));
+                if (hashToken == null) return string.Empty;
+                string[] hashTokens = hashToken.Split(':');
+                return hashTokens.Length < 2 ? string.Empty : hashTokens[1].Replace("'", string.Empty);
+            }
+        }
+        [JsonIgnore]
+        public string CollateralAddress
+        {
+            get
```

Comment:
There seems top be a lot of duplicated logic in these 3 `get`s. Please refactor to a method where you pass in the index that needs to be returned. I.e. 1 for CollateralAddress, 2 for PubKey etc.